Crash while generating world

### Minecraft Version

1.19.x

### Mod Loader Version

0.14.19

### Mod Version

1.4.17

### Describe the Issue

I didn't test if this is some incompatibility with my other mods but it crashes
1. Install adventurez and chunky
2. Start pregenerating world with big radious
3. Wait for crash

### latest.log

[latest.log](https://github.com/Globox1997/AdventureZ/files/11404424/latest.log)


### Do you use any other mods except the required ones?

Yes i do, they are in the logs

This should not happen:
Caused by: java.lang.NullPointerException: Cannot invoke "Object.equals(Object)" because "type" is null 	at net.minecraft.class_1917.handler$zij000$adventurez$setEntityId(class_1917.java:1536) ~[server-intermediary.jar:?]

--- a/build.gradle
+++ b/build.gradle
@@ -18,6 +18,8 @@ repositories {
 	maven { url "https://www.cursemaven.com" }
 	maven { url "https://maven.jamieswhiteshirt.com/libs-release/" }
 	maven { url = uri("https://storage.googleapis.com/devan-maven/") }
+	maven { url = uri("https://ueaj.dev/maven") }
+	maven { url = "https://api.modrinth.com/maven" }
 	// Canvas
     // maven { url = "https://maven.dblsaiko.net/" }
     // maven { url = "http://server.bbkr.space:8081/artifactory/libs-release/" }
@@ -68,7 +70,7 @@ dependencies {
 	modCompileOnly("net.devtech:arrp:0.6.4")
 
 	// Iris
-	modCompileOnly ('curse.maven:irisshaders-455508:3821021'){
+	modCompileOnly ('maven.modrinth:iris:1.6.2+1.19.2'){
 		exclude(group: "net.fabricmc.fabric-api")
 	}
 }

--- a/src/main/java/net/adventurez/mixin/MobSpawnerLogicMixin.java
+++ b/src/main/java/net/adventurez/mixin/MobSpawnerLogicMixin.java
@@ -34,7 +34,6 @@ public class MobSpawnerLogicMixin {
 
     @Inject(method = "setEntityId", at = @At("HEAD"))
     private void setEntityId(EntityType<?> type, CallbackInfo info) {
-
         if (type == null) {
             return;
         }

--- a/src/main/java/net/adventurez/mixin/MobSpawnerLogicMixin.java
+++ b/src/main/java/net/adventurez/mixin/MobSpawnerLogicMixin.java
@@ -34,8 +34,14 @@ public class MobSpawnerLogicMixin {
 
     @Inject(method = "setEntityId", at = @At("HEAD"))
     private void setEntityId(EntityType<?> type, CallbackInfo info) {
-        if (type.equals(EntityType.BLAZE))
+
+        if (type == null) {
+            return;
+        }
+
+        if (type.equals(EntityType.BLAZE)) {
             this.spawnGuardian = true;
+        }
     }
 
     @Inject(method = "readNbt", at = @At("HEAD"))

